### ISSUE TITLE
Access denied when creating symlinks on mounted volumes in mcr.microsoft.com/windows/servercore:ltsc2016

### ISSUE THREAD

**Issue by bob-du88** [moby/moby#41716]
**Description**
Access denied when creating symlinks on mounted volumes in mcr.microsoft.com/windows/servercore:ltsc2016

**Steps to reproduce the issue:**
1. Create a text file in c:\sym_test directory
   cd C:\sym_test\
   echo "this is symlink" > test.txt
2.Run powershell container and mount the working directory as volume
   docker run -it -v c:\sym_test:c:\sym_test mcr.microsoft.com/windows/servercore:ltsc2016 powershell

**Describe the results you received:**
Inside the container run
PS C:\> cd .\sym_test\
PS C:\sym_test> dir
    Directory: C:\sym_test
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       11/26/2020   4:49 PM             36 test.txt

PS C:\sym_test>
PS C:\sym_test>
PS C:\sym_test> cmd /c mklink sym_test.txt test.txt
Access is denied.

**it happens on binded volumes(c:\sym_test),but it's ok on unbinded volumes.**

PS C:\sym_test> cmd /c mklink c:\sym_test.txt test.txt
symbolic link created for c:\sym_test.txt <<===>> test.txt

**Describe the results you expected:**
creating a symbolic link success.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**
Client: Docker Engine - Community
 Azure integration  0.1.15
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:43:18 2020
 OS/Arch:           windows/amd64
 Experimental:      false

**Output of `docker info`:**
Client:
 Debug Mode: false

Server:
 Containers: 3
  Running: 2
  Paused: 0
  Stopped: 1
 Images: 35
 Server Version: 19.03.12
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 17763 (17763.1.amd64fre.rs5_release.180914-1434)
 Operating System: Windows 10 Enterprise LTSC 2019 Version 1809 (OS Build 17763.316)
 OSType: windows
 Architecture: x86_64
 CPUs: 8
 Total Memory: 7.79GiB
 Name: E0002136LT
 ID: WXVV:RIZA:DSXX:AEOD:DIZI:KAYQ:YGCS:2VQG:2JVH:UEIW:EWJI:UDAE
 Docker Root Dir: C:\ProgramData\Docker
 Debug Mode: true
  File Descriptors: -1
  Goroutines: 52
  System Time: 2020-11-26T16:39:25.8404499+08:00
  EventsListeners: 1
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

**Additional environment details (AWS, VirtualBox, physical, etc.):**
winver: Windows10 1809(OS 17763.316)

**Comment by bob-du88** [comment]
@thaJeztah do you encountered this problem?

**Comment by thaJeztah** [comment]
I'm not running docker on windows myself; does the same happen if the container is started with `process` isolation? (mostly trying to narrow down in what area to look for)

**Comment by ByronFaber** [comment]
Yes the problem occurs in both hyperv and process isolation modes.

**Comment by thaJeztah** [comment]
> Yes the problem occurs in both hyperv and process isolation modes.

Thanks! Trying to further narrow-down; does the same also happen on non bind-mounted volumes (so using a named or anonymous volumes?), e.g.;

```bash
docker run -it -v c:\sym_test mcr.microsoft.com/windows/servercore:ltsc2016 powershell
```
